pyro/util.py

```diff
@@ -205,7 +207,8 @@ def enum_extend(trace, msg, num_samples=None):
         num_samples = -1
 
     # Batched .enumerate_support() assumes batched values are independent.
-    batch_shape = msg["fn"].batch_shape(msg["value"], *msg["args"], **msg["kwargs"])
+    batch_shape = broadcast_shape(msg["fn"].shape(*msg["args"], **msg["kwargs"]),
+                                  msg["value"].size())[:-1]
```

Comment:
What is the `[:-1]` slice for?

Reply:
I seem to have missed updating this in the second part of the PR. Thanks for spotting this! Will change to using the `event_dim` arg and update.